File: 0988-smallest-string-starting-from-leaf/0988-smallest-string-starting-from-leaf.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def smallestFromLeaf(self, root: Optional[TreeNode]) -> str:
        def helper(root, cur):
            if not root:
                return
            cur = chr(ord('a')+root.val) + cur
            if root.left and root.right:
                return min(helper(root.left,cur),helper(root.right,cur))
            if root.left:
                return helper(root.left,cur)
            if root.right:
                return helper(root.right,cur)
            return cur
        return helper(root,"")

File: 0988-smallest-string-starting-from-leaf/0988-smallest-string-starting-from-leaf.cpp
#include<bits/stdc++.h>
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    string helper(TreeNode*root, string cur){
            cur = cur + string(1,'a'+root->val);
            if(root->left && root->right){
                return min(helper(root->left,cur),helper(root->right,cur));
            }
            if(root->left){
                return helper(root->left,cur);
            }
            if(root->right){
                return helper(root->right,cur);
            }
        reverse(cur.begin(),cur.end());
        return cur;
        }
    
    string smallestFromLeaf(TreeNode* root) {
        return helper(root, "");
    }
};